Revert/Sources/AutoLayoutMarginsViewController.swift

```diff
@@ -3,16 +3,51 @@
 
 import UIKit
 
-final class AutoLayoutMarginsViewController: RevertViewController {
-  override func viewDidLoad() {
-    super.viewDidLoad()
+private struct MarginValue {
+  mutating func increment() {
+    self.value = self.dynamicType.cappedMarginForMargin(self.value, delta: self.dynamicType.delta)
+  }
 
-    self.slider.value = 0
-    self.slider.minimumValue = 0
-    self.slider.maximumValue = 100
-    self.centerView.layoutMargins = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0)
+  mutating func decrement() {
+    self.value = self.dynamicType.cappedMarginForMargin(self.value, delta: -self.dynamicType.delta)
+  }
+
+  func layoutMarginsForCurrentValue() -> UIEdgeInsets {
+    return UIEdgeInsets(top: self.value, left: self.value, bottom: self.value, right: self.value)
   }
 
+  func currentValuePercentage() -> Float {
+    return Float(self.value / self.dynamicType.maxValue)
+  }
+
+  func isMaximumValue() -> Bool {
+    return self.value == self.dynamicType.maxValue
+  }
+
+  func isMinimumValue() -> Bool {
+    return self.value == self.dynamicType.minValue
+  }
+
+  // MARK: Private
+  private static let maxValue: CGFloat = 150
+  private static let minValue: CGFloat = 0
+  private static let delta: CGFloat = 10
+
+  private(set) var value: CGFloat = 0 {
+    didSet {
+      self.progress.completedUnitCount = Int64(self.value)
+    }
+  }
+
+  let progress = NSProgress(totalUnitCount:Int64(MarginValue.maxValue))
```

Comment:
Should `progress` be private too?  It is mixed in with the other `private` members.